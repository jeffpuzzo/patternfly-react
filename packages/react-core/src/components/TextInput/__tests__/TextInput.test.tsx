import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TextInput, TextInputBase } from '../TextInput';
import { ValidatedOptions } from '../../../helpers/constants';

const props = {
  onChange: jest.fn(),
  value: 'test input'
};

describe('TextInput', () => {
  test('input passes value and event to onChange handler', () => {
    render(<TextInputBase {...props} aria-label="test input" />);

    userEvent.type(screen.getByLabelText('test input'), 'new test input');
    expect(props.onChange).toHaveBeenCalled();
  });

  test('simple text input', () => {
    render(<TextInput {...props} aria-label="simple text input" />);
    expect(screen.getByLabelText('simple text input').outerHTML).toMatchSnapshot();
  });

  test('disabled text input', () => {
    render(<TextInput isDisabled aria-label="disabled text input" />);
    expect(screen.getByLabelText('disabled text input').outerHTML).toMatchSnapshot();
  });

  test('readonly text input', () => {
    render(<TextInput isReadOnly value="read only" aria-label="readonly text input" />);
    expect(screen.getByLabelText('readonly text input').outerHTML).toMatchSnapshot();
  });

  test('invalid text input', () => {
    render(<TextInput {...props} required validated={'error'} aria-label="invalid text input" />);
    expect(screen.getByLabelText('invalid text input').outerHTML).toMatchSnapshot();
  });

  test('validated text input success', () => {
    render(<TextInput {...props} required validated={ValidatedOptions.success} aria-label="validated text input" />);
    expect(screen.getByLabelText('validated text input').className).toContain('pf-m-success');
  });

  test('validated text input success', () => {
    render(<TextInput {...props} required validated={ValidatedOptions.warning} aria-label="validated text input" />);
    expect(screen.getByLabelText('validated text input').className).toContain('pf-m-warning');
  });

  test('validated text input', () => {
    render(<TextInput {...props} required validated={ValidatedOptions.error} aria-label="validated text input" />);
    expect(screen.getByLabelText('validated text input').outerHTML).toMatchSnapshot();
  });

  test('should throw console error when no aria-label, id or aria-labelledby is given', () => {
    const myMock = jest.fn();
    global.console = { ...global.console, error: myMock };

    render(<TextInput {...props} />);

    expect(myMock).toBeCalled();
  });

  test('should not throw console error when id is given but no aria-label or aria-labelledby', () => {
    const myMock = jest.fn();
    global.console = { ...global.console, error: myMock };

    render(<TextInput {...props} id="5" />);

    expect(myMock).not.toBeCalled();
  });

  test('should not throw console error when aria-label is given but no id or aria-labelledby', () => {
    const myMock = jest.fn();
    global.console = { ...global.console, error: myMock };

    render(<TextInput {...props} aria-label="test input" />);

    expect(myMock).not.toBeCalled();
  });

  test('should not throw console error when aria-labelledby is given but no id or aria-label', () => {
    const myMock = jest.fn();
    global.console = { ...global.console, error: myMock };

    render(<TextInput {...props} aria-labelledby="test input" />);

    expect(myMock).not.toBeCalled();
  });
});
