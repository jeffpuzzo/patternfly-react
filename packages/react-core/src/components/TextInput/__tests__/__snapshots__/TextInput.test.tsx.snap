// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput disabled text input 1`] = `"<input aria-label=\\"disabled text input\\" class=\\"pf-c-form-control\\" type=\\"text\\" aria-invalid=\\"false\\" disabled=\\"\\" data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TextInputBase-3\\" value=\\"\\">"`;

exports[`TextInput invalid text input 1`] = `"<input aria-label=\\"invalid text input\\" class=\\"pf-c-form-control\\" type=\\"text\\" aria-invalid=\\"true\\" data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TextInputBase-5\\" value=\\"test input\\">"`;

exports[`TextInput readonly text input 1`] = `"<input aria-label=\\"readonly text input\\" class=\\"pf-c-form-control\\" type=\\"text\\" aria-invalid=\\"false\\" readonly=\\"\\" data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TextInputBase-4\\" value=\\"read only\\">"`;

exports[`TextInput simple text input 1`] = `"<input aria-label=\\"simple text input\\" class=\\"pf-c-form-control\\" type=\\"text\\" aria-invalid=\\"false\\" data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TextInputBase-2\\" value=\\"test input\\">"`;

exports[`TextInput validated text input 1`] = `"<input aria-label=\\"validated text input\\" class=\\"pf-c-form-control\\" type=\\"text\\" aria-invalid=\\"true\\" data-ouia-component-type=\\"PF4/TextInput\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TextInputBase-8\\" value=\\"test input\\">"`;
